jenkins/scripts/run_tests.py

```
@@ -0,0 +1,705 @@
+    if os.path.exists(rerun_0_file):
+        print(f"Contents of {rerun_0_file}:")
+        print(Path(rerun_0_file).read_text())
+        print("There are some failed tests that cannot be rerun, skip the rerun step.")
+        return True, []
+
+    # Step 3: Count total failed tests
+    valid_count = 0
+    for times in [1, 2]:
+        rerun_file = os.path.join(rerun_dir, f"rerun_{times}.txt")
+        if os.path.exists(rerun_file):
+            lines = [l for l in Path(rerun_file).read_text().splitlines() if l.strip()]
```

Ruff (0.15.6)

[error] Ambiguous variable name: `l`

(E741)